### DISCUSSION TITLE
Iphone Saving Battery Mode not autoplaying video in DOM

### DISCUSSION THREAD

**Discussion by davidpino-gj** [cookpete/react-player#1779] [General]
Be sure to [search for your issue](https://github.com/CookPete/react-player/issues) before opening a new one.

#### Current Behavior
When my APP renders, the video from the MediaStream is being auto played and works just fine in all SO's and Devices, however, when my iPhone is on Saving Battery Mode, the video does not play and just shows a full blank container with a play button in the center (that doesn't work either)

![image](https://github.com/cookpete/react-player/assets/106541420/d2aca013-fb4e-4f6f-bf10-5c2e91dfab3b)

#### Expected Behavior
The video should autoplay just fine in Saving Battery Mode

![image](https://github.com/cookpete/react-player/assets/106541420/3535416b-083d-429f-8de7-ad77b97cf3bc)

#### Environment
- URL attempting to play: MediaStream
- Browser: Safari/Chrome
- Operating system: iOS 17

#### Other Information
Code Snippet: 

`<ReactPlayer
            url={mediaStream}
            height={'100%'}
            width={'100%'}
            controls={false}
            playing={true}
            muted={true}
            pip={isEnabled}
            playsinline={true}
            onEnablePIP={() => setEnabled(true)}
            onDisablePIP={() => {
              setEnabled(false)
              if (isIOS) {
                const videoTrack = CallService.getCurrentVideoTrack()
                const mediaStream = new MediaStream([videoTrack.getMediaStreamTrack()])
                agoraDispatch({
                  type: 'setMediaStream',
                  payload: {
                    mediaStream: null,
                  },
                })
                agoraDispatch({
                  type: 'setMediaStream',
                  payload: {
                    mediaStream,
                  },
                })
              }
            }}
            config={{
              file: {
                attributes: {
                  style: {
                    position: 'static',
                    objectFit: 'cover',
                    height: '100%',
                    width: '100%',
                  },
                },
              },
            }}
          />`
          
       If anyone could help me with this, I'd be very grateful. Thanks.

**Comment by davidpino-gj** [comment]
If I add the autoplay attribute (true or false) in my config prop, the play button dissapears but the issue remains.

`config={{
              file: {
                attributes: {
                  autoplay: true,
                  style: {
                    position: 'static',
                    objectFit: 'cover',
                    height: '100%',
                    width: '100%',
                  },
                },
              },
            }}`


![image](https://github.com/cookpete/react-player/assets/106541420/fe9c7b1c-1653-4a9b-bbeb-42c2f3a741e3)

**Comment by AlbertSmit** [comment]
This is by Apple's design in iOS' Safari; I don't think there is a way around it. A user interaction is required to start the video.
